### PR TITLE
Adds framework for equip delays and adds it to the straight jacket

### LINKED ISSUE #28355: [consistency] You can straightjacket yourself instantly but can't take it off
You can instantly equip a straight jacket like any suit but this immediately restrains your hands. Good luck getting it off anytime soon. A delay like with cuffing is needed to prevent self-restraining instantly. 
Tested with hot keys (did it by accident the other say but forgot to report it)
Might also apply to cuff shoes.

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
add: The straight jacket now takes five seconds to put on.
/:cl:

Items now have a variable, called `equip_delay_self`, checked when a mob attempts to _manually_ equip an item. This means it _won't call on force-equips like_ `equip_to_slot_or_del`.

To make it more clear, `put_on_delay` has also been renamed to `equip_delay_other`.

This is currently only implemented in the form of straight jackets, which have a five-second delay before equipping them; this is to both fix a consistency issue and to prevent you from accidentally straight-jacketing yourself, which, while hilarious, isn't very fun, especially when an Unknown with a gas mask drags you helplessly away into maintenance to show you the light of Ratvar.

Closes #28355.

### CODE REVIEW

**Comment by calzilla1** [comment]
I hear salt

**Comment by Xhuis** [comment]
How do you mean?

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -59,6 +59,7 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 	var/armour_penetration = 0 //percentage of armour effectiveness to remove
 	var/list/allowed = null //suit storage stuff.
 	var/obj/item/device/uplink/hidden_uplink = null
+	var/equip_delay = 0 //In deciseconds, how long an item takes to equip; counts only for normal clothing slots, not pockets etc.
 	var/strip_delay = 40
 	var/put_on_delay = 20
```
Comment (RemieRichards) [inline code/game/objects/items.dm]: *LOUD COUGH*

Comment (kevinz000) [inline code/game/objects/items.dm]: lel

Comment (Cyberboss) [inline code/game/objects/items.dm]: @xhuis is actually in the right here. That delay is for the strip panel.

File: code/game/objects/items.dm [hunk 2]
```diff
@@ -407,11 +408,11 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 //if this is being done by a mob other than M, it will include the mob equipper, who is trying to equip the item to mob M. equipper will be null otherwise.
 //If you are making custom procs but would like to retain partial or complete functionality of this one, include a 'return ..()' to where you want this to happen.
 //Set disable_warning to 1 if you wish it to not give you outputs.
-/obj/item/proc/mob_can_equip(mob/M, mob/equipper, slot, disable_warning = 0)
+/obj/item/proc/mob_can_equip(mob/M, mob/equipper, slot, disable_warning = 0, bypass_equip_delay)
```
Comment (Cyberboss) [inline code/game/objects/items.dm]: Put = FALSE as a default

File: code/modules/mob/living/carbon/human/inventory.dm [hunk 3]
```diff
@@ -1,5 +1,5 @@
-/mob/living/carbon/human/can_equip(obj/item/I, slot, disable_warning = 0)
-	return dna.species.can_equip(I, slot, disable_warning, src)
+/mob/living/carbon/human/can_equip(obj/item/I, slot, disable_warning = 0, bypass_equip_delay)
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/inventory.dm]: = FALSE

File: code/game/objects/items.dm (hunk 1 above)
Comment (RemieRichards) [inline code/game/objects/items.dm]: The point is that the names aren't helpful, the name ```put_on_delay``` will now confusingly conflict with ```equip_delay```, despite the fact it MEANS "put on someone else delay"

File: code/modules/mob/living/carbon/human/species.dm [hunk 4]
```diff
@@ -603,7 +603,7 @@
 	// handles the equipping of species-specific gear
 	return
 
-/datum/species/proc/can_equip(obj/item/I, slot, disable_warning, mob/living/carbon/human/H)
+/datum/species/proc/can_equip(obj/item/I, slot, disable_warning, mob/living/carbon/human/H, bypass_equip_delay)
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/species.dm]: = FALSE

**Comment by RemieRichards** [comment]
Not a refactor, it's an addition.

File: code/modules/mob/living/carbon/human/inventory.dm (hunk 3 above)
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/inventory.dm]: Also change the 0 to FALSE

File: code/game/objects/items.dm (hunk 2 above)
Comment (Cyberboss) [inline code/game/objects/items.dm]: Also change the 0 to FALSE

Comment (Cyberboss) [inline code/game/objects/items.dm]: mob/living for both M and equipper

File: code/modules/mob/mob.dm [hunk 5]
```diff
@@ -198,9 +198,9 @@
 //set qdel_on_fail to have it delete W if it fails to equip
 //set disable_warning to disable the 'you are unable to equip that' warning.
 //unset redraw_mob to prevent the mob from being redrawn at the end.
-/mob/proc/equip_to_slot_if_possible(obj/item/W, slot, qdel_on_fail = 0, disable_warning = 0, redraw_mob = 1)
+/mob/proc/equip_to_slot_if_possible(obj/item/W, slot, qdel_on_fail = 0, disable_warning = 0, redraw_mob = 1, bypass_equip_delay)
```
Comment (Cyberboss) [inline code/modules/mob/mob.dm]: 0s to FALSE, 1 to TRUE

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items.dm (hunk 1 above)
Comment (octareenroon91) [inline code/game/objects/items.dm]: `equip_delay_self` and `equip_delay_other` might be preferable, but then we'd need to change every file with a `put_on_delay`.

Would closely matching the other name, such as with `put_on_self_delay`, help??

Comment (RemieRichards) [inline code/game/objects/items.dm]: either work, former preferred to latter.

Comment (Xhuis) [inline code/game/objects/items.dm]: oh no, you've got a cold! here, have some water to soothe your cough.
💧

Comment (RemieRichards) [inline code/game/objects/items.dm]: Don't test me.

Comment (Xhuis) [inline code/game/objects/items.dm]: I'm just trying to be lighthearted. ☹️

**Comment by Xhuis** [comment]
I renamed `put_on_delay` to `equip_delay_other` and replaced all of its occurrences. I also renamed `equip_delay` to `equip_delay_self`.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**